src/core/ai/systemPrompt/contributors.ts

```diff
@@ -23,3 +25,101 @@ export class DynamicContributor implements SystemPromptContributor {
         return this.promptGenerator(context);
     }
 }
+
+export interface FileContributorOptions {
+    includeFilenames?: boolean | undefined;
+    separator?: string | undefined;
+    errorHandling?: 'skip' | 'placeholder' | 'error' | undefined;
+    maxFileSize?: number | undefined;
+    encoding?: string | undefined;
+    includeMetadata?: boolean | undefined;
+}
+
+export class FileContributor implements SystemPromptContributor {
+    constructor(
+        public id: string,
+        public priority: number,
+        private files: string[],
+        private options: FileContributorOptions = {}
+    ) {}
+
+    async getContent(_context: DynamicContributorContext): Promise<string> {
+        const {
+            includeFilenames = true,
+            separator = '\n\n---\n\n',
+            errorHandling = 'skip',
+            maxFileSize = 100000,
+            encoding = 'utf-8',
+            includeMetadata = false,
+        } = this.options;
+
+        const fileParts: string[] = [];
+
+        for (const filePath of this.files) {
+            try {
+                // Resolve relative paths
+                const resolvedPath = resolve(filePath);
+
+                // Check if file is .md or .txt
+                const ext = extname(resolvedPath).toLowerCase();
+                if (ext !== '.md' && ext !== '.txt') {
+                    if (errorHandling === 'error') {
+                        throw new Error(`File ${filePath} is not a .md or .txt file`);
+                    } else if (errorHandling === 'placeholder') {
+                        fileParts.push(`[File ${filePath} skipped: not a .md or .txt file]`);
+                    }
+                    continue;
+                }
+
+                // Check file size
+                const stats = await stat(resolvedPath);
+                if (stats.size > maxFileSize) {
+                    if (errorHandling === 'error') {
+                        throw new Error(
+                            `File ${filePath} exceeds maximum size of ${maxFileSize} bytes`
+                        );
+                    } else if (errorHandling === 'placeholder') {
+                        fileParts.push(
+                            `[File ${filePath} skipped: exceeds maximum size of ${maxFileSize} bytes]`
+                        );
+                    }
+                    continue;
+                }
+
+                // Read file content
+                const content = await readFile(resolvedPath, { encoding: encoding as any });
```

Comment:
we can probably just hardcode utf-8, don't think encoding needs tobe customizable atm